src/main/java/io/temporal/internal/replay/ReplayWorkflowExecutor.java

```diff
@@ -295,32 +190,27 @@ private void eventLoop() {
       failure = workflow.mapUnexpectedException(e);
       completed = true;
     }
-  }
-
-  private void mayBeCompleteWorkflow() {
     if (completed) {
       completeWorkflow();
-    } else {
-      updateTimers();
     }
   }
 
   private void completeWorkflow() {
     if (failure != null) {
-      commandHelper.failWorkflowExecution(failure);
+      workflowStateMachines.newFailWorkflow(failure.getFailure());
       metricsScope.counter(MetricsType.WORKFLOW_FAILED_COUNTER).inc(1);
     } else if (cancelRequested) {
-      commandHelper.cancelWorkflowExecution();
+      workflowStateMachines.newCancelWorkflow();
       metricsScope.counter(MetricsType.WORKFLOW_CANCELLED_COUNTER).inc(1);
```

Comment:
Canceled vs Cancelled

Reply:
It looks like we are using Canceled in the API: https://github.com/temporalio/api/search?q=canceled&unscoped_q=canceled

[Google says ](https://www.google.com/search?q=canceled+vs+cancelled&oq=CANCELED+V&aqs=chrome.1.69i57j0l7.2643j0j7&sourceid=chrome&ie=UTF-8) that Canceled is American and cancelled is British.

Reply:
I'm going to rename cancelled to canceled everywhere.